starknet_py/tests/e2e/account/outside_execution_test.py

```diff
@@ -5,33 +5,49 @@
 from starknet_py.constants import ANY_CALLER, OutsideExecutionInterfaceID
 from starknet_py.hash.selector import get_selector_from_name
 from starknet_py.net.account.account import BaseAccount
-from starknet_py.net.client_models import Call, OutsideExecutionTimeBounds
+from starknet_py.net.client_models import (
+    Call,
+    OutsideExecutionTimeBounds,
+    ResourceBounds,
+)
 from starknet_py.tests.e2e.fixtures.constants import MAX_FEE
 from starknet_py.transaction_errors import TransactionRevertedError
 
 
 @pytest.mark.asyncio
 async def test_argent_account_outside_execution_compatibility(
     argent_account: BaseAccount,
+    argent_account_v040: BaseAccount,
 ):
     result = await argent_account.supports_interface(OutsideExecutionInterfaceID.V1)
     assert result is True
     result = await argent_account.supports_interface(OutsideExecutionInterfaceID.V2)
     assert result is False
 
+    result = await argent_account_v040.supports_interface(
+        OutsideExecutionInterfaceID.V1
+    )
+    assert result is True
+    result = await argent_account_v040.supports_interface(
+        OutsideExecutionInterfaceID.V2
+    )
+    assert result is True
```

Comment:
Could we change this test to be parametrized?

Reply:
I'm afraid it's not trivial due to dependency injection. I attempted to reduce amount of statements. please check.